dist/legacy-assets/js/main.js

```diff
@@ -55782,15 +55782,47 @@ function viewReleaseSelector() {
 
 function viewReportDetails(collection, isPublished, $this) {
 
-    var details, reportDetails, published, events;
+    // Enum type of filterable data
+    const filterableReportType = {
+        'dataset': 0,
+        'datasetVersion': 1
+    };
+
+    // Retrieve all information for a specific type of filterable data report
+    function getFilterableReportData(filterableDataTypeToUse, fullDatasetCollectionData) {
```

Comment:
This function has been written with DRY and Open Close principles in mind.
Currently the only data used is the URI, however a usecase is present to expand this function and add Duration and size, some fields might be mandatory for all types of dataset reports whereas some might be specific to an edition, dataset, version or other.

The outer function `getFilterableReportData` is used for a specific type of filterable data report whereas the `getMandatoryFilterableReportInformation` is for mandatory data that is in all types of filterable reports.